pennylane_honeywell/device.py

```diff
@@ -128,15 +164,184 @@ def set_api_configs(self):
         """
         Set the configurations needed to connect to HQS API.
         """
-        self._api_key = self._api_key or os.getenv("HQS_TOKEN")
-        if not self._api_key:
-            raise ValueError("No valid api key for HQS platform found.")
-        self.header = {
-            "User-Agent": "pennylane-honeywell_v{}".format(__version__),
-            self.API_HEADER_KEY: self._api_key,
-        }
+        self._user = self._user or os.getenv("HQS_USER")
+
+        if not self._user:
+            # Check the PennyLane config file too
+            config = qml.default_config
+            config.safe_get(config._config, *["honeywell", "global", "user_email"])
+
         self.hostname = "/".join([self.BASE_HOSTNAME, self.TARGET_PATH])
 
+    @staticmethod
+    def token_is_expired(token):
+        """Check whether a given token is expired.
+
+        Args:
+            token (str): A token to check, could be an access token or a
+                refresh token. The token is decoded using JWT to check its
+                expiry date.
+
+        Returns:
+            bool: whether the token is expired or not
+        """
+        try:
+            token_expiry_time = jwt.decode(token, verify=False, algorithms=["RS256"])["exp"]
```

Comment:
The signature is not being checked if we set `verify=False`, so no `jwt.ExpiredSignatureError` would be raised here. The server could potentially send back something invalid, so put this decoding into a `try-except`.

`verify=False` was an option used in PyJWT 1.71, indeed instead of  `options={"verify_exp": False}`. Some issues were reported for the latest versions of PyJWT, but just gave it a try and it seems to work :+1: